tpm2/structures.go

```diff
@@ -371,3 +371,158 @@ type tpmtSigScheme struct {
 	Scheme Algorithm
 	Hash   Algorithm
 }
+
+// AttestationData contains data attested by TPM commands (like Certify).
+type AttestationData struct {
+	Magic               uint32
+	Type                tpmutil.Tag
+	QualifiedSigner     Name
+	ExtraData           []byte
+	ClockInfo           ClockInfo
+	FirmwareVersion     uint64
+	AttestedCertifyInfo *CertifyInfo
+}
+
+// DecodeAttestationData decode a TPMS_ATTEST message. No error is returned if
+// in has extra trailing data.
+func DecodeAttestationData(in []byte) (*AttestationData, error) {
+	buf := bytes.NewBuffer(in)
+
+	var ad AttestationData
+	if err := tpmutil.UnpackBuf(buf,
+		&ad.Magic,
+		&ad.Type,
+	); err != nil {
+		return nil, fmt.Errorf("decoding Magic/Type: %v", err)
+	}
+	n, err := decodeName(buf)
+	if err != nil {
+		return nil, fmt.Errorf("decoding QualifiedSigner: %v", err)
+	}
+	ad.QualifiedSigner = *n
+	if err := tpmutil.UnpackBuf(buf,
+		&ad.ExtraData,
+		&ad.ClockInfo,
+		&ad.FirmwareVersion,
+	); err != nil {
+		return nil, fmt.Errorf("decoding ExtraData/ClockInfo/FirmwareVersion: %v", err)
+	}
+
+	// Add support for more types as needed.
```

Comment:
Please clarify this comment. Is this meant to be a TODO? Or is this meant to point out to the reader that this is one of the places that would need to be changed if more types were needed?

The reason I have trouble with this comment is that it's written in the imperative, which is not what I normally expect from a non-TODO comment.

Reply:
Expanded the comment a bit.
It's not a TODO because I don't actually plan to implement it. But it's a suggestion for future reader, in case the need arises.